drivers/gpu/drm/vc4/vc4_txp.c

```diff
@@ -272,6 +272,13 @@ static int vc4_txp_connector_atomic_check(struct drm_connector *conn,
 		return -EINVAL;
 	}
 
+	if (conn_state->rotation == (DRM_MODE_ROTATE_0 | DRM_MODE_TRANSPOSE) &&
```

Comment:
Not sure if I'm violating the API, but I've been using the transpose feature by just setting `DRM_MODE_TRANSPOSE` without explicitly or'ing `DRM_MODE_ROTATE_0` as not rotating seems to be the default when setting the rotation property to `0`. As such my old code still slipped by this condition. With the explicit  `DRM_MODE_ROTATE_0` also or'ed, the commit fails as expected.

Reply:
Yes it's violating the API. Added an extra patch that should validate that.